mirdata/irmas.py

```diff
@@ -0,0 +1,406 @@
+        force_overwrite=force_overwrite,
+        cleanup=cleanup,
+    )
+
+
+def validate(data_home=None, silence=False):
+    """Validate if the stored dataset is a valid version
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        missing_files (list): List of file paths that are in the dataset index
+            but missing locally
+        invalid_checksums (list): List of file paths that file exists in the dataset
+            index but has a different checksum compare to the reference checksum
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    missing_files, invalid_checksums = utils.validator(
+        DATA.index, data_home, silence=silence
+    )
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    """Return track ids
+
+    Returns:
+        (list): A list of track ids
+    """
+    return list(DATA.index.keys())
+
+
+def load(data_home=None):
+    """Load IRMAS dataset
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        (dict): {`track_id`: track data}
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    data = {}
+    for key in DATA.index.keys():
+        data[key] = Track(key, data_home=data_home)
+    return data
+
+
+def load_pred_inst(audio_path, annotation_path, train):
+    """Load predominant instrument of track
+
+    Args:
+        audio_path (str): Local path where the track is stored.
+        train (bool): Flag to know if track is from the train or test set
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+        annotation_path (str): Local path where the testing annotation is stored.
+    Returns:
+        pred_inst (str): track predominant instrument extracted from filename
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+
+    duration = None
+    if audio_path is not None:
+        if os.path.exists(audio_path):
+            duration = librosa.get_duration(filename=audio_path)
+
+    if train is True:
+        split_1 = audio_path.split('[')[1]
+        pred_inst_code = split_1.split(']')[0]
+
+        return utils.EventData(
+            np.array([0]), np.array([duration]), np.array([pred_inst_code])
+        )
+
+    else:
+        with open(annotation_path, 'r') as fopen:
+            pred_inst_file = fopen.readlines()
+            pred_inst = []
+            for inst_ in pred_inst_file:
+                inst_code = inst_[:3]
+                pred_inst.append(inst_code)
+
+        return utils.EventData(
+            np.array([0]), np.array([duration]), np.array([tuple(pred_inst)])
+        )
```

Comment:
I know we went back and forth with this and sorry if it is confusing! Finally we converged that if the annotation doesn't change over time, we will keep it same way we keep metadata. So if you had only one instrument, you would return a string, and in this case that you could get multiple, I guess you would have a list of strings? Does this make sense?

Reply:
Ok! It does make sense, parsing the data will turn more straightforward. So should I add the instrument annotation to _track_metadata? I will implement int this way and see how it looks like!

Reply:
So, the issue here is the following: if I want to return the instrument annotation as a String, there is no argument in jams_converter that accepts it, apart from adding it to the metadata together with 'genre' and 'drum'. 
However, although I can retrieve the predominant instrument annotation from the training samples filename in the index, I need to access all the .txt files of the testing samples annotations to retrieve the data. For that reason, when trying to add the predominant instrument to the metadata, the pytest fails because it requires the whole dataset to pass. That is why, storing `predominant_instrument` as `EventData` was working well, because I could use the function to load the instrument annotation in the `@utils.cached_property`. If I try to load the instrument annotation directly from the Track class in order to add it to the metadata, the pytests are failing because I need the whole dataset locally (or at least that what I understand). Am I correct? Did I explain myself?

Reply:
Mmmm... this is because of using the index in the `_load_metadata` right? I think it might be solved with my other comment in the implementation of that function

Reply:
I would say that this is because of parsing annotation files (in this case the .txt files) in the Track class itself. I mean, running the `load_pred_inst` function in the Track class to parse the intrument annotation and insert it into `_track_metadata` is not passing the tests, especially the following one: `pyest tests/irmas.py --local`. I believe that doing it inside `_load_metadata` would fail as well, but I haven't tried tbh. I will take a deep look into the other comment about `LargeData` to try to solve this! I am not completely sure about that assumptions... at the end you are way more familiar than me with mirdata...!!!

Reply:
But I am saying that because when running pytests on IRMAS dataset, if I parse information in the Track class, I get an error because the loader is looking for the whole set of annotations. Note that _predominant instrument_ is the only annotation in this dataset that is written in a single .txt file for each testing track. How could we add it into the metadata?

Reply:
Could we insert the instrument data as "tag open"? That is what we are doing for the [Mridangam Stroke Dataset](https://github.com/mir-dataset-loaders/mirdata/pull/285/) (that's the link of the Mridangam Stroke PR). This dataset has algo an annotation that lasts the same for the whole track!